### PR TITLE
Improve Theme Editor Dialogs

### LINKED ISSUE #29038: Oddly inconsistent UI in theme editor
**Godot version:**
3.1.1 official

**OS/device including version:**
Windows 10 64 bit

**Issue description:**
Massive vertical whitespace, should probably use a dropdown menu instead of the odd two dots.
image

Two menus use the two dots while the third uses the more intuitive dropdown.
image

### PR DESCRIPTION
**Pull request by swarnimarun**
Fixes https://github.com/godotengine/godot/issues/29038.

I made the changes from the fix and some others as well. Like removing the excess space and make the name be after data type cause it makes more sense. 
As it's options will be dependent on what we select in data type.

### CODE REVIEW

File: editor/plugins/theme_editor_plugin.cpp [hunk 1]
```diff
@@ -56,14 +56,10 @@ void ThemeEditor::_refresh_interval() {
 	_propagate_redraw(main_container);
 }
 
-void ThemeEditor::_type_menu_cbk(int p_option) {
-
-	type_edit->set_text(type_menu->get_popup()->get_item_text(p_option));
-}
```
Comment (akien-mga) [inline editor/plugins/theme_editor_plugin.cpp]: There are code style issues due to the two consecutive newlines here and where `_name_menu_cbk` was removed.

Comment (akien-mga) [reply editor/plugins/theme_editor_plugin.cpp]: And trailing whitespace: https://travis-ci.org/godotengine/godot/jobs/535167849

Comment (swarnimarun) [reply editor/plugins/theme_editor_plugin.cpp]: forgot clang-format :P

Comment (swarnimarun) [reply editor/plugins/theme_editor_plugin.cpp]: Ok now it should be fine.

**Comment by swarnimarun** [comment]
Part of #29063

**Comment by YeldhamDev** [comment]
@swarnimarun The "Name" menu should also be consistent with the others (radio box, same width as the button, and capitalized). It also should be disabled when empty.

Also, this removes the ability to write custom strings in the "Type" and "Name" sections. I never used that feature, but if it was there, there was probably a reason why.

**Comment by swarnimarun** [comment]
@YeldhamDev yep it does but I am not really sure how that will even be used at the moment. If there is importance I could do it.

As for the Name Menu, I am not sure, probably it's a good idea but why the radio box?

**Comment by YeldhamDev** [comment]
@akien-mga Do you have recommendations on who to query on the subject of custom names?

@swarnimarun About the radio box, it just makes sense as it's a menu where you select on item. It should also have something select to begin with.

**Comment by akien-mga** [comment]
> Do you have recommendations on who to query on the subject of custom names?

I'm not sure what those custom names are, but I'd ask `git blame`. If it turns out to be there unmodified since 2014, either ask reduz or consider that nobody knows nor uses them :P

**Comment by YeldhamDev** [comment]
Yep:
Screenshot_20190527_222115
@reduz Does this even do anything?!

**Comment by swarnimarun** [comment]
@YeldhamDev Fixed the one about Radiobutton actually it was a misunderstanding on my part I thought you meant for me to add Radio Button to it explicitly the mistake was actually just that I didn't change the add_item to add to the OptionMenu but let it add to the popup menu... :P 

Sorry for that idiotic mistake also fixed the size being reset while at it.

**Comment by swarnimarun** [comment]
I will let it rest until we get the custom names sorted for now   :3

**Comment by YeldhamDev** [comment]
The "Name" menu only updates when clicking on it, it should also update when the "Type" and "Data Type" menus are changed as well.

And the items in the "Name" menu should be capitalized, as stated above.

**Comment by YeldhamDev** [comment]
Reduz has informed me that custom theme constants should stay.

**Comment by akien-mga** [comment]
> Reduz has informed me that custom theme constants should stay.

Bump.

**Comment by swarnimarun** [comment]
O.o Seems like I forgot to finish this. Oops. Will finish it ASAP.

**Comment by swarnimarun** [comment]
Fixed it. 
Also @YeldhamDev the problem with the capitalization is not of consistency but of what the engine reads. 
The Engine tags are all snake_case and thus keeping them the same is the better option as it would mean breaking in engine consistency.

image

**Comment by YeldhamDev** [comment]
I see, but there's still some more things:
- The selected type being reset to "Button" should be avoided (if possible).
- Closing and reopening the popup causes the "Name" section to not update properly depending of the previous selected options (likely caused by point above):
Peek 2019-06-19 21-24
Peek 2019-06-19 21-25

**Comment by swarnimarun** [comment]
The reset bug really feels new... (I need to see what I messed up) :/ Will fix it.

**Comment by YeldhamDev** [comment]
@swarnimarun Any update on this?

**Comment by swarnimarun** [comment]
I didn't forget this time, I just didn't have time to move it along as the bug seemingly needs a bit more attention then I had thought. :P 
I will probably redo it and force push the changes. ASAP

**Comment by YeldhamDev** [comment]
Bump.

**Comment by swarnimarun** [comment]
Gonna redo, but not sure when so closing...

**Pull request closed without merging**